daphne_worker/src/dap.rs

```diff
@@ -219,48 +220,124 @@ impl<'srv> BearerTokenProvider<'srv> for DaphneWorker<'srv> {
     fn is_taskprov_leader_bearer_token(&self, token: &BearerToken) -> bool {
         self.get_global_config().allow_taskprov
             && match &self.config().taskprov {
-                Some(config) => config.leader_bearer_token == *token,
+                Some(config) => config.leader_auth.as_ref() == token,
                 None => false,
             }
     }
 
     fn is_taskprov_collector_bearer_token(&self, token: &BearerToken) -> bool {
         self.get_global_config().allow_taskprov
             && match &self.config().taskprov {
-                Some(config) => config.collector_bearer_token == *token,
+                Some(config) => {
+                    config
+                        .collector_auth
+                        .as_ref()
+                        .expect("collector authorization method not set")
+                        .as_ref()
+                        == token
+                }
                 None => false,
             }
     }
 }
 
 #[async_trait(?Send)]
-impl DapAuthorizedSender<BearerToken> for DaphneWorker<'_> {
+impl DapAuthorizedSender<DaphneWorkerAuth> for DaphneWorker<'_> {
     async fn authorize(
         &self,
         task_id: &Id,
         media_type: &'static str,
         _payload: &[u8],
-    ) -> std::result::Result<BearerToken, DapError> {
-        Ok(self
-            .authorize_with_bearer_token(task_id, media_type)
-            .await?
-            .value()
-            .clone())
+    ) -> std::result::Result<DaphneWorkerAuth, DapError> {
+        // TODO Add support for authorizing the request with TLS client certificates:
+        // https://developers.cloudflare.com/workers/runtime-apis/mtls/
+        Ok(DaphneWorkerAuth::BearerToken(
+            self.authorize_with_bearer_token(task_id, media_type)
+                .await?
+                .value()
+                .clone(),
+        ))
     }
 }
 
 #[async_trait(?Send)]
-impl<'srv, 'req> DapAggregator<'srv, 'req, BearerToken> for DaphneWorker<'srv>
+impl<'srv, 'req> DapAggregator<'srv, 'req, DaphneWorkerAuth> for DaphneWorker<'srv>
 where
     'srv: 'req,
 {
     type WrappedDapTaskConfig = GuardedDapTaskConfig<'req>;
 
     async fn authorized(
         &self,
-        req: &DapRequest<BearerToken>,
+        req: &DapRequest<DaphneWorkerAuth>,
     ) -> std::result::Result<bool, DapError> {
-        self.bearer_token_authorized(req).await
```

Comment:
I have not had a chance to test this, but will do so before we merge. We could also consider writing E2E tests for this (I believe wrangler dev can be configured to expect TLS client certs), but that would take some doing.